### DISCUSSION TITLE
PAG内置报so 错误

### DISCUSSION THREAD

**Discussion by zhongbingrong** [Tencent/libpag#2291] [General]
【版本信息】
PAG SDK 版本: Android SDK完整版 4.3.56
操作系统版本: Android 14

【预期的情况】
全部开机播放动画

【实际的情况】
内容在Android的系统应用SystemUI中，crash

05-13 16:56:35.805  3694  3694 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 3694 (ndroid.systemui), pid 3694 (ndroid.systemui)
05-13 16:56:36.057  3770  3770 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
05-13 16:56:36.057  3770  3770 F DEBUG   : Build fingerprint: 'coolpad/CP36t/CP36t:14/CP36t.20240513.0S.SE/20240513:userdebug/release-keys'
05-13 16:56:36.057  3770  3770 F DEBUG   : Revision: '0'
05-13 16:56:36.057  3770  3770 F DEBUG   : ABI: 'arm64'
05-13 16:56:36.057  3770  3770 F DEBUG   : Timestamp: 2024-05-13 16:56:35.871909231+0800
05-13 16:56:36.057  3770  3770 F DEBUG   : Process uptime: 1s
05-13 16:56:36.057  3770  3770 F DEBUG   : Cmdline: com.android.systemui
05-13 16:56:36.057  3770  3770 F DEBUG   : pid: 3694, tid: 3694, name: ndroid.systemui  >>> com.android.systemui <<<
05-13 16:56:36.057  3770  3770 F DEBUG   : uid: 10147
05-13 16:56:36.057  3770  3770 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
05-13 16:56:36.057  3770  3770 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
05-13 16:56:36.057  3770  3770 F DEBUG   : Abort message: 'JNI DETECTED ERROR IN APPLICATION: java_class == null
05-13 16:56:36.057  3770  3770 F DEBUG   :     in call to GetStaticMethodID
05-13 16:56:36.057  3770  3770 F DEBUG   :     from void org.libpag.PAGPlayer.nativeInit()'
05-13 16:56:36.057  3770  3770 F DEBUG   :     x0  0000000000000000  x1  0000000000000e6e  x2  0000000000000006  x3  0000007ff68a87b0
05-13 16:56:36.057  3770  3770 F DEBUG   :     x4  fefefefefefefeff  x5  fefefefefefefeff  x6  fefefefefefefeff  x7  7f7f7f7f7f7f7f7f
05-13 16:56:36.057  3770  3770 F DEBUG   :     x8  00000000000000f0  x9  0000007098edb050  x10 0000000000000001  x11 0000007098f24b50
05-13 16:56:36.057  3770  3770 F DEBUG   :     x12 000000000000fa94  x13 00000000000000b7  x14 0000007ff68a75b0  x15 000000002dacf86b
05-13 16:56:36.057  3770  3770 F DEBUG   :     x16 0000007098f90cf8  x17 0000007098f6d460  x18 0000006ca396f4a8  x19 0000000000000e6e
05-13 16:56:36.057  3770  3770 F DEBUG   :     x20 0000000000000e6e  x21 00000000ffffffff  x22 0000000000000058  x23 0000006dfa2b9439
05-13 16:56:36.057  3770  3770 F DEBUG   :     x24 0000000000000058  x25 0000000000000001  x26 0000000000000000  x27 0000006dfae2c000
05-13 16:56:36.057  3770  3770 F DEBUG   :     x28 b400006e6bc2f730  x29 0000007ff68a8830
05-13 16:56:36.057  3770  3770 F DEBUG   :     lr  0000007098f15178  sp  0000007ff68a8790  pc  0000007098f151a4  pst 0000000000001000
05-13 16:56:36.057  3770  3770 F DEBUG   : 98 total frames
05-13 16:56:36.057  3770  3770 F DEBUG   : backtrace:
05-13 16:56:36.057  3770  3770 F DEBUG   :       #00 pc 000000000005c1a4  /apex/com.android.runtime/lib64/bionic/libc.so (abort+164) (BuildId: 19217ea6661b6085ec146e3216f02b35)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #01 pc 0000000000771b00  /apex/com.android.art/lib64/libart.so (art::Runtime::Abort(char const*)+904) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #02 pc 00000000000357d0  /apex/com.google.mainline.primary.libs@341177000/lib64/libbase.so/75d3253827fcfd7a8d7b02ad45991611ec4ca424c0278e13e8acfad4d14e597a3ecff6c0caa2b785c73838528ee6e9c2b313240ff895f50ee39b1d7bc10f390a/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80) (BuildId: 6f67f69ff36b970d0b831cfdab3b578d)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #03 pc 0000000000034d58  /apex/com.google.mainline.primary.libs@341177000/lib64/libbase.so/75d3253827fcfd7a8d7b02ad45991611ec4ca424c0278e13e8acfad4d14e597a3ecff6c0caa2b785c73838528ee6e9c2b313240ff895f50ee39b1d7bc10f390a/libbase.so (android::base::LogMessage::~LogMessage()+352) (BuildId: 6f67f69ff36b970d0b831cfdab3b578d)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #04 pc 00000000003ec9b0  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbort(char const*, char const*)+3540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #05 pc 0000000000525794  /apex/com.android.art/lib64/libart.so (art::JNI<false>::GetStaticMethodID(_JNIEnv*, _jclass*, char const*, char const*)+392) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #06 pc 00000000000d3f10  /system/lib64/libpag.so (BuildId: fc46a68cc9bd68d77ee21226db9f074832a29012)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #07 pc 00000000000e0b24  /system/lib64/libpag.so (BuildId: fc46a68cc9bd68d77ee21226db9f074832a29012)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #08 pc 0000000000355830  /apex/com.android.art/lib64/libart.so (art_quick_generic_jni_trampoline+144) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #09 pc 00000000005b9798  /apex/com.android.art/lib64/libart.so (nterp_helper+152) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #10 pc 000000000010657c  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (org.libpag.PAGPlayer.<clinit>+20)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #11 pc 000000000033f080  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #12 pc 000000000024b364  /apex/com.android.art/lib64/libart.so (art::ClassLinker::InitializeClass(art::Thread*, art::Handle<art::mirror::Class>, bool, bool)+3304) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #13 pc 00000000003350e0  /apex/com.android.art/lib64/libart.so (NterpAllocateObject+1676) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #14 pc 00000000005bf7b0  /apex/com.android.art/lib64/libart.so (nterp_allocate_object+48) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #15 pc 00000000005b1dd4  /apex/com.android.art/lib64/libart.so (nterp_op_new_instance+84) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #16 pc 00000000001072dc  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (org.libpag.PAGView.b+208)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #17 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #18 pc 0000000000106fe6  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (org.libpag.PAGView.<init>+58)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #19 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #20 pc 00000000000d195c  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.yulong.keyguard.YLKeyguardChargingView.initPAGView+112)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #21 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #22 pc 00000000000d17ba  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.yulong.keyguard.YLKeyguardChargingView.<init>+90)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #23 pc 000000000033eda4  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+612) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #24 pc 00000000005978b4  /apex/com.android.art/lib64/libart.so (art::Constructor_newInstance0(_JNIEnv*, _jobject*, _jobjectArray*) (.__uniq.114100828369361150012403296395805223021)+812) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #25 pc 000000000009e3f8  /system/framework/arm64/boot.oat (art_jni_trampoline+104) (BuildId: 54cb69b1f07cfce37533b3d1612536f7c8ff66c6)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #26 pc 0000000000271098  /system/framework/arm64/boot.oat (java.lang.reflect.Constructor.newInstance+56) (BuildId: 54cb69b1f07cfce37533b3d1612536f7c8ff66c6)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #27 pc 000000000075fb1c  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.createView+1612) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #28 pc 00000000007607f8  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.createViewFromTag+1304) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #29 pc 000000000075eb10  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.parseInclude+1808) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.057  3770  3770 F DEBUG   :       #30 pc 0000000000761b28  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.rInflate+920) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #31 pc 000000000075ec8c  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.parseInclude+2188) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #32 pc 0000000000761b28  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.rInflate+920) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #33 pc 0000000000761134  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.inflate+996) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #34 pc 0000000000760cc4  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.inflate+212) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #35 pc 0000000000760bd4  /system/framework/arm64/boot-framework.oat (android.view.LayoutInflater.inflate+52) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #36 pc 00000000005ba6b0  /apex/com.android.art/lib64/libart.so (nterp_helper+4016) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #37 pc 00000000003b1a3a  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get2+78)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #38 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #39 pc 00000000003aa798  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get+80)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #40 pc 0000000002000480  /memfd:jit-cache (deleted) (offset 0x2000000) (dagger.internal.DoubleCheck.get+176)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #41 pc 00000000005bb4d4  /apex/com.android.art/lib64/libart.so (nterp_helper+7636) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #42 pc 00000000003b1a68  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get2+124)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #43 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #44 pc 00000000003aa798  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get+80)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #45 pc 0000000002000480  /memfd:jit-cache (deleted) (offset 0x2000000) (dagger.internal.DoubleCheck.get+176)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #46 pc 00000000005bb4d4  /apex/com.android.art/lib64/libart.so (nterp_helper+7636) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #47 pc 00000000003b1bf2  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get2+518)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #48 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #49 pc 00000000003aa798  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get+80)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #50 pc 0000000002000480  /memfd:jit-cache (deleted) (offset 0x2000000) (dagger.internal.DoubleCheck.get+176)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #51 pc 00000000005bb4d4  /apex/com.android.art/lib64/libart.so (nterp_helper+7636) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #52 pc 00000000003abf48  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get0+5976)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #53 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #54 pc 00000000003aa7ac  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get+100)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #55 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #56 pc 00000000000da24a  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (dagger.internal.DoubleCheck.get+26)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #57 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #58 pc 00000000003ac408  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get0+7192)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #59 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #60 pc 00000000003aa7ac  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.dagger.DaggerReferenceGlobalRootComponent$ReferenceSysUIComponentImpl$SwitchingProvider.get+100)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #61 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #62 pc 00000000000da24a  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (dagger.internal.DoubleCheck.get+26)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #63 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #64 pc 000000000034843e  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.accessibility.SystemActions.registerActions+314)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #65 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #66 pc 0000000000348728  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.accessibility.SystemActions.start+240)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #67 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #68 pc 0000000000345162  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIApplication.startStartable+50)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #69 pc 00000000005b9734  /apex/com.android.art/lib64/libart.so (nterp_helper+52) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #70 pc 0000000000344a02  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIApplication$$ExternalSyntheticLambda1.run+110)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #71 pc 00000000005bb474  /apex/com.android.art/lib64/libart.so (nterp_helper+7540) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #72 pc 00000000003451b6  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIApplication.timeInitialization+54)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #73 pc 00000000005b9734  /apex/com.android.art/lib64/libart.so (nterp_helper+52) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #74 pc 000000000034507e  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIApplication.startServicesIfNeeded+214)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #75 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #76 pc 0000000000344f90  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIApplication.startServicesIfNeeded+88)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #77 pc 00000000005ba654  /apex/com.android.art/lib64/libart.so (nterp_helper+3924) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #78 pc 00000000003460ea  /system_ext/priv-app/CPSystemUI/CPSystemUI.apk (com.android.systemui.SystemUIService.onCreate+18)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #79 pc 0000000000387510  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread.handleCreateService+1744) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #80 pc 0000000000379c2c  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread$H.handleMessage+6924) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #81 pc 00000000005d9d98  /system/framework/arm64/boot-framework.oat (android.os.Handler.dispatchMessage+152) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #82 pc 00000000005dcfa4  /system/framework/arm64/boot-framework.oat (android.os.Looper.loopOnce+980) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #83 pc 00000000005dcb44  /system/framework/arm64/boot-framework.oat (android.os.Looper.loop+548) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #84 pc 000000000038f388  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread.main+1640) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #85 pc 000000000033f080  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #86 pc 00000000003884dc  /apex/com.android.art/lib64/libart.so (_jobject* art::InvokeMethod<(art::PointerSize)8>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1588) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #87 pc 0000000000387e98  /apex/com.android.art/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*) (.__uniq.165753521025965369065708152063621506277)+32) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #88 pc 00000000000a1ed8  /system/framework/arm64/boot.oat (art_jni_trampoline+120) (BuildId: 54cb69b1f07cfce37533b3d1612536f7c8ff66c6)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #89 pc 00000000007c7804  /system/framework/arm64/boot-framework.oat (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+116) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #90 pc 00000000007d17ec  /system/framework/arm64/boot-framework.oat (com.android.internal.os.ZygoteInit.main+3164) (BuildId: efb4f9319e4a50b72883be24ebff94044bc30cfe)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #91 pc 000000000033f080  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #92 pc 00000000004e1ea8  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeWithVarArgs<_jmethodID*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+728) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.058  3770  3770 F DEBUG   :       #93 pc 000000000057b930  /apex/com.android.art/lib64/libart.so (art::JNI<true>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+156) (BuildId: 735f12f804f88d62a2cb437261076ff7)
05-13 16:56:36.059  3770  3770 F DEBUG   :       #94 pc 00000000000dbca8  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+104) (BuildId: 0dd18b398d7eb410d9f33f441d35b7dc)
05-13 16:56:36.059  3770  3770 F DEBUG   :       #95 pc 00000000000e7e7c  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+940) (BuildId: 0dd18b398d7eb410d9f33f441d35b7dc)
05-13 16:56:36.059  3770  3770 F DEBUG   :       #96 pc 000000000000254c  /system/bin/app_process64 (main+1260) (BuildId: 8af507f6acd9563ed93bb2be0b5e70ec)
05-13 16:56:36.059  3770  3770 F DEBUG   :       #97 pc 00000000000546e8  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+104) (BuildId: 19217ea6661b6085ec146e3216f02b35)

**Comment by zhongbingrong** [comment]
[Uploading a.txt…]()

**Reply by zhongbingrong** [reply]
完整Log如a.txt，内置在Android系统应用SystemUI中，开机一直crash.

**Comment by zhongbingrong** [comment]
降SDK的版本 4.0.5.79 就没有问题了

**Reply by kevingpqi123** [reply]
如果你们使用的是 aar，感觉是代码混淆的问题，可以看下代码混淆规则：
https://pag.art/docs/sdk.html#android-%E7%AB%AF%E6%8E%A5%E5%85%A5

pag 在开机动画中有应用的案例，此类问题如果还存在建议 google 搜索下，和 sdk 的关系应该不大

**Reply by zhongbingrong** [reply]
注意： 需要在混淆列表里面，添加 libpag 的 keep 规则：

    -keep class org.libpag.** {*;}
    -keep class androidx.exifinterface.** {*;}

混淆规则是加过的，之前用的4.0.5.79版本没有问题，这是Android 14用的最新的SDK，报的是so的出错。Android系统应用使用pag，需要集成aar和so两部分。上面的是报错，还请帮忙看下。

**Reply by kevingpqi123** [reply]
主要我们缺乏复现场景，由于人力有限，这类问题需要有明确的可以复现的场景我们才好分析，可以看下怎么构造复现场景，你们可以基于 pag 的官方 demo 做改造，有复现路径后我们再跟进，原则上我们只保证官方 demo 是正常运行的

**Reply by zhongbingrong** [reply]
你们那边应该无法复现这样的场景，官方的demo只运行在三方应用中，运行在系统源码中，如果可以你们就分析一下提供的Log，谢谢，需要什么支持我也可以协助，主要是低版本是没有这个问题，最新版本有这样的问题，如果无法觖就关闭吧，我们暂时先用稳定版本，谢谢您的支持~
